src/Algebra/Construct/DirectProduct.agda

```diff
@@ -151,3 +161,73 @@ abelianGroup G H = record
     ; comm = λ x y → (G.comm , H.comm) <*> x <*> y
     }
   } where module G = AbelianGroup G; module H = AbelianGroup H
+
+semiringWithoutAnnihilatingZero : SemiringWithoutAnnihilatingZero a ℓ₁ →
+                                  SemiringWithoutAnnihilatingZero b ℓ₂ →
+                                  SemiringWithoutAnnihilatingZero (a ⊔ b) (ℓ₁ ⊔ ℓ₂)
+semiringWithoutAnnihilatingZero R S = record
+  { isSemiringWithoutAnnihilatingZero = record
+      { +-isCommutativeMonoid = CommutativeMonoid.isCommutativeMonoid
+                                  (commutativeMonoid R.+-commutativeMonoid
+                                                     S.+-commutativeMonoid)
+      ; *-isMonoid = Monoid.isMonoid (monoid R.*-monoid S.*-monoid)
+      ; distrib    = distribˡ , distribʳ
```

Comment:
This can be written inline as:
```
distrib = (λ x y z → (R.distribˡ , S.distribˡ) <*> x <*> y <*> z)
        , (λ x y z → (R.distribʳ , S.distribʳ) <*> x <*> y <*> z)
```